File: Udemy/Udemy/Models/Category.cs
using System;
using System.Collections.Generic;

namespace Udemy.Models;

public partial class Category
{
    public int CategoryId { get; set; }

    public string CategoryName { get; set; } = null!;

    public int? ParentCategoryId { get; set; }

    public int? CreatedBy { get; set; }

    public int? ModifiedBy { get; set; }

    public DateTime? CreatedDate { get; set; }

    public DateTime? ModifiedDate { get; set; }

    public virtual ICollection<Course> Courses { get; set; } = new List<Course>();

    public virtual User? CreatedByNavigation { get; set; }

    public virtual ICollection<Category> InverseParentCategory { get; set; } = new List<Category>();

    public virtual User? ModifiedByNavigation { get; set; }

    public virtual Category? ParentCategory { get; set; }
}

File: Frontend/src/app/Interfaces/category.ts
export interface Category {
    categoryId: number,
    categoryName: string,
    parentCategoryId: null|number,
    subCourses: Array<Category>
}
export interface CategoryDto{
    categoryName: string,
    parentCategoryId: number|null
}
export interface Categorysub {
    categoryId: number,
    categoryName: string,
    parentCategoryId: null|number,
}